### DISCUSSION TITLE
How to create a jagged array from a flat list and the layout/num values

### DISCUSSION THREAD

**Discussion by mattbellis** [scikit-hep/awkward#623] [Q&A]
On the [tutorials site](https://awkward-array.org/)?

In the [Python docstrings](https://awkward-array.readthedocs.io/)?

In the [C++ doxygen API](https://awkward-array.readthedocs.io/en/latest/_static/index.html)?

What, specifically, needs more explanation?

I'd like to create an awkward array from a flat list of numbers and then the layout/num information. For example, can I get this

```
<Array [[1, 2], [3, 4, 5, 6]] type='2 * var * int64'>
```

from 
```
vals = [1, 2, 3, 4, 5, 6]
num = [2,4]
```

**Comment by jpivarski** [comment]
Okay, this needs more explanation (as a tutorial?). Meanwhile, the place to look for your specific question is [ak.unflatten](https://awkward-array.readthedocs.io/en/latest/_auto/ak.unflatten.html).

**Comment by mattbellis** [comment]
Ah, this is perfect! I feel like I should have been able to intuit the function name. 😁 Much thanks!

**Comment by jpivarski** [comment]
We had some discussion about the function name (@nsmith- and I) and I agree that it's not the best. It's only something that you'd think of if you had previously gone in the other direction (flatten, then unflatten).

With a long to-do list, it's been hard for me to get to writing documentation. My first goal is to get the stubs in the left-bar of [awkward-array.org](https://awkward-array.org) filled out. On the other hand, answering questions as they come up organically might actually make more difference in the long run. Although I'll make sure to include "unflatten" in the docs, I'll convert this request for documentation into a Q&A discussion, so that it's always prominent (not "hidden," like closed issues).

**Reply by mattbellis** [reply]
No, I think you misunderstood my sincerity! It's a good name! The opposite of flatten! I actually think I should have tried that or tried looking for that name! There was no snark intended!

Commenting is hard. Sorry for the confusion of my intent.

**Reply by jpivarski** [reply]
(I also didn't think you were being snarky. But we did have misgivings about the name, myself included. "unflatten" was the best we came up with, and maybe it's better than we thought.)
